src/domain/entities/risk-assessment/RiskAssessmentGrading.ts

```diff
@@ -0,0 +1,169 @@
+    type: CapacityOptionsTypes;
+    weight: 1 | 2 | 3;
+};
+export const LowCapacity: CapacityOptions = {
+    type: "ProvincialNationalLevel",
+    weight: 1,
+};
+export const MediumCapacity: CapacityOptions = {
+    type: "ProvincialLevel",
+    weight: 2,
+};
+export const HighCapacity: CapacityOptions = {
+    type: "NationalInternationalLevel",
+    weight: 3,
+};
+
+export type Grade = "Grade1" | "Grade2" | "Grade3";
+
+type AllOptionTypes =
+    | WeightedOptionTypes
+    | PopulationWeightOptionsTypes
+    | GeographicalSpreadOptionsTypes
+    | CapacityOptionsTypes
+    | Grade;
+
+const translations: Record<AllOptionTypes, string> = {
+    Low: i18n.t("Low"),
+    Medium: i18n.t("Medium"),
+    High: i18n.t("High"),
+    LessPercentage: i18n.t("Less than 0.1%"),
+    MediumPercentage: i18n.t("Between 0.1% to 0.25%"),
+    HighPercentage: i18n.t("Above 0.25%"),
+    WithinDistrict: i18n.t("Within a district"),
+    MoretThanOneDistrict: i18n.t("Within a province with more than one district affected"),
+    MoreThanOneProvince: i18n.t(
+        "More than one province affected with high threat of spread locally and internationally"
+    ),
+    ProvincialNationalLevel: i18n.t(
+        "Available within the district with support from provincial and national level"
+    ),
+    ProvincialLevel: i18n.t(
+        "Available within the province with minimal support from national level"
+    ),
+    NationalInternationalLevel: i18n.t(
+        "Available at national with support required from international"
+    ),
+    Grade1: i18n.t("Grade 1"),
+    Grade2: i18n.t("Grade 2"),
+    Grade3: i18n.t("Grade 3"),
+};
+
+interface RiskAssessmentGradingAttrs extends Ref {
+    lastUpdated: Date;
+    populationAtRisk: PopulationWeightOptions;
+    attackRate: WeightedOptions;
+    geographicalSpread: GeographicalSpreadOptions;
+    complexity: WeightedOptions;
+    capacity: CapacityOptions;
+    reputationalRisk: WeightedOptions;
+    severity: WeightedOptions;
+}
+
+export class RiskAssessmentGrading extends Struct<RiskAssessmentGradingAttrs>() {
+    private constructor(attrs: RiskAssessmentGradingAttrs) {
+        super(attrs);
+    }
+
+    public static create(attrs: RiskAssessmentGradingAttrs): RiskAssessmentGrading {
+        return new RiskAssessmentGrading(attrs);
+    }
+
+    public static getTranslatedLabel(key: AllOptionTypes): string {
+        return translations[key];
+    }
+
+    getGrade = (): Either<Error, Grade> => {
+        return this.calculateGrade();
+    };
+
+    calculateGrade(): Either<Error, Grade> {
+        const totalWeight =
+            this.populationAtRisk.weight +
+            this.attackRate.weight +
+            this.geographicalSpread.weight +
+            this.complexity.weight +
+            this.capacity.weight +
+            this.reputationalRisk.weight +
+            this.severity.weight;
+
+        if (totalWeight > 21) return Either.error(new Error(i18n.t("Invalid grade")));
```

Comment:
Do we want the validation here, which will fire on `getGrade`, or we do the validation at create, and prevent an invalid object to be build? (it depends on how this will be used). In the latter case, `grade` would be a normal get property.

Reply:
i think we can do the validation on getGrade and show an error and give the customer the opportunity to re-enter values for other properties and re-trigger getGrade calculation.